src/clr-angular/forms/combobox/options.ts

```diff
@@ -1,59 +1,140 @@
 /**
- * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Component, ElementRef, Inject, Injector, OnDestroy, Optional } from '@angular/core';
-import { Subscription } from 'rxjs';
-
-import { AbstractPopover } from '../../popover/common/abstract-popover';
-import { Point } from '../../popover/common/popover';
+import {
+  Component,
+  ContentChildren,
+  ElementRef,
+  Inject,
+  Input,
+  Optional,
+  QueryList,
+  AfterViewInit,
+  OnDestroy,
+} from '@angular/core';
+import { DOCUMENT } from '@angular/common';
 import { POPOVER_HOST_ANCHOR } from '../../popover/common/popover-host-anchor.token';
+import { OptionSelectionService } from './providers/option-selection.service';
+import { IF_ACTIVE_ID } from '../../utils/conditional/if-active.service';
+import { AriaService } from '../../utils/aria/aria.service';
+import { ClrCommonStringsService } from '../../utils/i18n/common-strings.service';
+import { LoadingListener } from '../../utils/loading/loading-listener';
+import { ClrLoadingState } from '../../utils/loading/loading';
+import { ClrOption } from './option';
+import { ComboboxFocusHandler } from './providers/combobox-focus-handler.service';
+import { fromEvent, Subscription } from 'rxjs';
+import { ClrPopoverToggleService } from '../../utils/popover/providers/popover-toggle.service';
+
+let nbOptionsComponents = 0;
+
+@Component({
+  selector: 'clr-options',
+  template: `
+    <div *ngIf="optionSelectionService.loading" class="clr-options-loading">
```

Comment:
This needs a View Basics test for both the loading state and the loaded state.

Reply:
Your IDE probably double-posted this. 
Test added, as commented previously.